cli/src/semgrep/engine.py

```diff
@@ -33,52 +33,54 @@ class EngineType(Enum):
     @classmethod
     def decide_engine_type(
         cls,
-        requested_engine: Optional["EngineType"] = None,
+        logged_in: bool = False,
+        engine_flag: Optional["EngineType"] = None,
+        run_secrets: bool = False,
+        interfile_diff_scan_enabled: bool = False,
+        # ci-only args
         scan_handler: Optional[ScanHandler] = None,
         git_meta: Optional[GitMeta] = None,
-        run_secrets: bool = False,
-        enable_pro_diff_scan: bool = False,
         supply_chain_only: bool = False,
     ) -> "EngineType":
         """Select which Semgrep engine type to use if none is explicitly requested.
 
         Considers settings from Semgrep Cloud Platform and version control state.
         """
-        # Change default to pro-engine intrafile if secrets was requested.
-        # Secrets is built into pro-engine, but any pro-setting should work.
-        if (
-            not (scan_handler and scan_handler.deepsemgrep)
-            and requested_engine is None
-            and run_secrets
-        ):
-            requested_engine = cls.PRO_LANG
-        elif run_secrets and requested_engine is cls.OSS:
-            # Should be impossible if the CLI gates impossible arguement combinations.
-            raise SemgrepError("Semgrep Secrets is not part of the open source engine")
+        interfile_is_requested_via_app = scan_handler and scan_handler.deepsemgrep
 
-        if git_meta and scan_handler:
-            if scan_handler.deepsemgrep and requested_engine is None:
+        requested_engine = engine_flag
+        if engine_flag is None:
+            if interfile_is_requested_via_app:
                 requested_engine = cls.PRO_INTERFILE
-
-            if (
-                requested_engine == cls.PRO_INTERFILE
-                and not git_meta.is_full_scan
-                and not enable_pro_diff_scan
-            ):
+            elif run_secrets:
                 requested_engine = cls.PRO_INTRAFILE
 
-        # Using PRO_LANG engine since PRO_INTERFILE/PRO_INTRAFILE defaults to -j 1
-        # note if using OSS, then will keep using OSS
+        if run_secrets and engine_flag is cls.OSS:
+            # Should be impossible if the CLI gates impossible argument combinations.
+            # TODO Can we delete this check?
+            raise SemgrepError("Semgrep Secrets is not part of the open source engine")
+
+        diff_scan = git_meta and not git_meta.is_full_scan
         if (
-            requested_engine in {cls.PRO_INTERFILE, cls.PRO_INTRAFILE}
-            and supply_chain_only
+            diff_scan
+            and not interfile_diff_scan_enabled
```

Comment:
Wasn't this enabled by default for everyone now?

Reply:
I think it's still under controlled rollout? Anyway, would rather remove that in a separate PR